src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,505 @@
+            private final List<StrandedInterval> distalTargets;
+
+            OverlapAndCoherenceIterator(final BreakpointEvidence evidence,
+                                        final SVIntervalTree<List<BreakpointEvidence>> evidenceTree,
+                                        final ReadMetadata readMetadata,
+                                        final int minEvidenceMapQ) {
+                this.readMetadata = readMetadata;
+                this.minEvidenceMapQ = minEvidenceMapQ;
+                treeItr = evidenceTree.overlappers(evidence.getLocation());
+                isForwardStrand = evidence.isEvidenceUpstreamOfBreakpoint();
+                checkCoherence = (isForwardStrand != null) && evidence.hasDistalTargets(readMetadata, minEvidenceMapQ);
+                distalTargets = checkCoherence ?
+                        new ArrayList<>(evidence.getDistalTargets(readMetadata, minEvidenceMapQ)) : null;
+            }
+
+            @Override
+            public boolean hasNext() {
+                if ( listItr != null && listItr.hasNext() ) {
+                    return true;
+                }
+                while(treeItr.hasNext()) {
+                    listItr = treeItr.next().getValue().iterator();
+                    if(listItr.hasNext()) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            @Override
+            public ImmutablePair<BreakpointEvidence, Boolean> next() {
+                if ( !hasNext() ) {
+                    throw new NoSuchElementException("No next element.");
+                }
+                final BreakpointEvidence overlapper = listItr.next();
+                Boolean isCoherent = false;
+                if(checkCoherence) {
+                    Boolean overlapperStrand = overlapper.isEvidenceUpstreamOfBreakpoint();
+                    if(overlapperStrand == isForwardStrand && overlapper.hasDistalTargets(readMetadata, minEvidenceMapQ)) {
+                        for(final StrandedInterval distalTarget : distalTargets) {
+                            for(final StrandedInterval overlapperDistalTarget : overlapper.getDistalTargets(readMetadata, minEvidenceMapQ)) {
+                                if(distalTarget.getStrand() == overlapperDistalTarget.getStrand()
+                                        && distalTarget.getInterval().overlaps(overlapperDistalTarget.getInterval())) {
+                                    isCoherent = true;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+                return new ImmutablePair<>(overlapper, isCoherent);
+            }
+        }
+    }
+    */
+
+    private static class UnscaledOverlapInfo {
+        final int numOverlap;
+        final int numCoherent;
+        final int overlapMappingQuality;
```

Comment:
Since you are summing the mapping qualities here I'd call this field `totalOverlapperMappingQuality` or something like that. Also, why not move computing the mean mapping quality to here, since it's got the sum and the number of overlappers.

Reply:
Renamed totalOverlapMappingQuality, and moved calculation of meanOverlapMappingQuality into UnscaledOverlapInfo.

Reply:
Can you also rename `coherentMappingQuality` to `totalCoherentMappingQuality`? I think I keep looking at those and getting tripped up forgetting that they are actually sums of mapping qualities rather than means or maximums.